src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1799,10 +1799,14 @@ private Symbol GetDeclaredMember(NamespaceOrTypeSymbol container, TextSpan decla
                     zeroWidthMatch = symbol;
                 }
 
-                // Handle the case of the implementation of a partial method.
-                var partial = symbol.Kind == SymbolKind.Method
-                    ? ((MethodSymbol)symbol).PartialImplementationPart
-                    : null;
+                // Handle the case of the implementation of a partial member.
+                Symbol partial = symbol switch
+                {
+                    MethodSymbol method => method.PartialImplementationPart,
+                    SourcePropertySymbol property => property.PartialImplementationPart,
+                    _ => null
+                };
```

Comment:
This PR includes some tests for `GetDeclaredSymbol` public API. Do we have tests for `GetDeclaredSymbol` on the parameters of a partial indexer? 
It looks like there is additional logic in this file that looks at `PartialDefinitionPart` for partial methods for that scenario, which might need to be updated to account for partial properties.

Reply:
I have addressed the remaining usages of PartialDefinition/ImplementationPart in this file.

There is one other usage of PartialDefinitionPart which doesn't need to account for properties, because it is for a method type parameter scenario, and properties don't have type parameters.